### PR TITLE
Adding dotenv for root .env functionality

### PR DESCRIPTION
**Pull request by chriswedgwood**
### What change does this PR introduce?
This is adding python-dotenv so we can add environment variables into a .env file in the root of the project for those who cant use docker. It also improves the getting started documentation around the db setup.

### Notes

You will need to add a .env file to the root of the project. This is in .gitignore so is not part of the project
put this in it:

DATABASE_URL=postgres://sharespots:sharespots@localhost:5432/sharespots
SECRET_KEY=Ajdasdasjd98as7yd8wa67982yq9hr20j0qd9uqu89i

### CODE REVIEW

File: sharespots/settings/base.py [hunk 1]
```diff
@@ -6,8 +6,15 @@
     get_env,
 )
 
+from dotenv import load_dotenv
+
```
Comment (tsvetelinak0) [inline sharespots/settings/base.py]: load_dotenv() -- working fine when adding this line as per docs / showing error w\ it

File: README.md [hunk 2]
```diff
@@ -142,19 +142,19 @@ Type "help" for help.
...
 Before running migrate/runserver you will need to add the environment variable
-DATABASE_URL into your system.
-
+DATABASE_URL into your system. To do this add a .env file into the root of your project
+and add the following two lines:
 <pre>
 
-$ export DATABASE_URL=postgres://sharespots:<i>your_password_not_this</i>@127.0.0.1:5432/sharespots
-
+DATABASE_URL=postgres://sharespots:sharespots@loclahost:5432/sharespots
```
Comment (tsvetelinak0) [inline README.md]: DATABASE_URL=postgres://sharespots:sharespots@localhost:5432/sharespots

little typo on this line.. should be localhost

File: requirements.txt [hunk 3]
```diff
@@ -9,3 +9,4 @@ django-heroku==0.3.1
 django-extensions==2.1.6
 jupyter==1.0.0
 Werkzeug==0.14.1  # https://github.com/pallets/werkzeug
+python-dotenv==0.10.1 # https://github.com/joke2k/django-environ
```
[the remark this answers is not in the document]
Comment (chriswedgwood) [reply requirements.txt]: Thanks Tsvetsi yes wrong url. Will change

File: sharespots/settings/base.py (hunk 1 above)
Comment (chriswedgwood) [reply sharespots/settings/base.py]: What error are you getting?

File: README.md (hunk 2 above)
Comment (chriswedgwood) [reply README.md]: Change :)

File: sharespots/settings/base.py (hunk 1 above)
Comment (tsvetelinak0) [reply sharespots/settings/base.py]: this is the err without load_dotenv() django.core.exceptions.ImproperlyConfigured: The SECRET_KEY setting must not be empty.

Comment (chriswedgwood) [reply sharespots/settings/base.py]: I have the followoing  lines that run load_env:
if os.path.exists(ENV_FILE):
    load_dotenv(dotenv_path=ENV_FILE,override=True)

Can you debug and check if your code is making it past the if statement. 

This setup works for me on windows but haven't confirmed on my mac

Comment (tsvetelinak0) [reply sharespots/settings/base.py]: Everything works fine now!

**Comment by abu271** [comment]
I closed this becasue it's a stale pull request and `DATABASE_URL` and `SECRET_KEY` already exist in the .env folder.

**Pull request closed without merging**